gpu/octree/src/cuda/octree_host.cu

```diff
@@ -100,6 +102,50 @@ namespace
         return count;
     } 
 
+    std::pair<uint3, std::uint8_t> nearestVoxel(const OctreeImpl::PointType& query, const unsigned& level, const std::uint8_t& mask, const float3& minp, const float3& maxp, const uint3& index)
+    {
+        assert(mask != 0);
+        //identify closest voxel
+        float closest_distance = std::numeric_limits<float>::max();
+        unsigned closest_index = 0;
+        uint3 closest = make_uint3(0,0,0);
+        const unsigned voxel_width = 1 << (level + 2);
```

Comment:
can this be interpreted as a width? isn't it more like a width scaling factor?

Reply:
Although I understand why you put this here, to avoid recomputing things over and over for each loop, there's a strong chance this will be optimized, so you can keep move this variable in a more reduced scope.

But let's try in compiler explorer and see what happens.

Reply:
https://godbolt.org/z/7rqoMx

the functions produce the same assembly because of the constness of the variables involved.